src/student_success_tool/analysis/pdp/targets/shared.py

```diff
@@ -280,6 +280,72 @@ def _compute_intensity_num_terms(
         for intensity, time, unit in intensity_time_lefts
     ]
 
+# THIS IS A NEW / MODIFIED FUNCTION TO GET THE TERM REPRESENTING END OF FIRST YEAR FOR A STUDENT, including pre-cohort cols
+
+def get_first_student_year_term(
```

Comment:
since the check-in for Rutgers is at the end of student's first year - I grabbed the second record, not the 1st. This is to ensure we're only keeping students who actually completed a full first-year (I'm assuming students who didn't wouldn't have a second record). And yes, we want the second term of the first year no matter what that term is. Does that make sense / seem sensible or am I missing something/another way to do it/misinterpreting the ask?

Reply:
Hey, as I've said before, this function's name (and its docstring) is actively misleading: You're selecting the student's second term, regardless of whether the student had three terms in their first year, or if they had pre-cohort terms before their first year, or some other edge case. If you don't care about these cases, you should make that clear in the function name and docs: something like `get_nth_student_term(df, n, ...)`, and then you can use `n=1` for your use case. (Be sure to note that `n=0` => "first", etc.)

Reply:
done!